vowpalwabbit/gd.h

```diff
@@ -81,8 +81,8 @@ inline void foreach_feature(vw& all, example& ec, R& dat)
 inline float inline_predict(vw& all, example& ec)
 {
   return all.weights.sparse ?
-    inline_predict<sparse_parameters>(all.weights.sparse_weights, all.ignore_some_linear, all.ignore_linear, all.interactions, all.permutations, ec, ec.l.simple.initial) :
-    inline_predict<dense_parameters>(all.weights.dense_weights, all.ignore_some_linear, all.ignore_linear, all.interactions, all.permutations, ec, ec.l.simple.initial);
+    inline_predict<sparse_parameters, v_array<v_string>>(all.weights.sparse_weights, all.ignore_some_linear, all.ignore_linear, all.interactions, all.permutations, ec, ec.l.simple.initial) :
```

Comment:
I'd like to get rid of the extra type parameterization by coalescing between the vw_slim_predict and vw.  Can you modify vw to use vector instead here?

Reply:
done

Reply:
done